pkg/render/apiserver.go

```diff
@@ -541,6 +541,15 @@ rules:
 func (c *apiServerComponent) apiServer() *appsv1.Deployment {
 	var replicas int32 = 1
 
+	hostNetwork := false
+	if c.installation.Spec.KubernetesProvider == operator.ProviderEKS &&
+		c.installation.Spec.CNI.Type == operator.PluginCalico {
+		// Workaround the fact that webhooks don't work for non-host-networked pods
+		// when in this networking mode on EKS, because the control plane nodes don't run
+		// Calico.
+		hostNetwork = true
```

Comment:
Should we have a field in the API that indicates this?

Reply:
I don't think so - this is really a workaround for a limitation in the combination of:
- How EKS control plane does not allow pod scheduling
- How Calico networking doesn't make pod IPs routable to instances outside of the cluster.

It's not something that can / should be user-configurable, it's just required to make the user's provided intent "work".